File: solutions/objective-c/difference-of-squares/1/DifferenceOfSquares.m
#import <Foundation/Foundation.h>
#include <math.h>
#import "DifferenceOfSquares.h"
@implementation DifferenceOfSquares
- (id)initWithMax:(int)max {
    if (self = [super init]) {
        maxValue = max;
    }
    return self;
}
-(int)squareOfSum {
    int value = 0;
    for (int i = 1; i <= maxValue; i++) {
        value = i + value;
    }
    return pow(value, 2);
}
-(int)sumOfSquares {
    int value = 0;
    for (int i = 1; i <= maxValue; i++) {
        value = pow(i, 2) + value;
    }
    return value;
}
-(int)differenceOfSquares {
    return [self squareOfSum] - [self sumOfSquares];
}
@end

File: solutions/delphi/difference-of-squares/1/difference_of_squares.pas
unit uDifferenceOfSquares;

interface

function squareOfSum(AInp : integer) : integer;

function sumOfSquares(AInp : integer) : integer;

function differenceOfSquares(AInp : integer) : integer;

implementation

function squareOfSum(AInp : integer) : integer;
var
  I: Integer;
begin
  Result :=1;
  for I := 2 to AInp do
    Result := Result + i;
  Result := sqr(Result);
end;

function sumOfSquares(AInp : integer) : integer;
var
  I: Integer;
begin
  Result := 1;
  for I := 2 to AInp do
    Result := Result + sqr(i);
end;

function differenceOfSquares(AInp : integer) : integer;
begin
  Result := squareOfSum(AInp) - sumOfSquares(AInp);
end;

end.